blottertrax/description_provider.py

```diff
@@ -41,8 +41,20 @@ def get_reply(self, parsed_submission: ParsedSubmission) -> str:
 
         life_span = '' if has_life_span is False else '({} to {})'.format(life_span_begin, life_span_end)
         tags = ', '.join(map(lambda t: t['name'], artist['tag-list'][:5])) if has_tags else 'none'
-        socials = ', '.join(map(lambda u: '[{}]({})'.format(u['type'], u['target']),
-                                artist['url-relation-list'])) if has_socials else 'none'
+        socials = map(lambda u: '[{}]({})'.format(u['type'], '\)'.join(u['target'].split(')'))),
+                                artist['url-relation-list']) if has_socials else 'none'
+        if socials != 'none':
```

Comment:
I think we should define this as a function and call it from the map function directly so we only loop over everything once.

`    
    socials = map(lambda u: self.format_network_to_friendly_name(artist, u)) if has_socials else 'none'
`

....

    @classmethod
    def format_network_to_friendly_name(cls, artist, info) -> str:
        social_networks = ["twitter.com", "facebook.com", "instagram.com"]
        target = info['target']
        network_type = info['type']
        if network_type == 'social network':
            for network in social_networks:
                if network in target:
                    network_type = network
        return '[{}]({})'.format(network_type, '\)'.join(target.split(')')), artist['url-relation-list'])


Similar to this maybe (untested).

Reply:
Tested as well as my code then, hold on